### DISCUSSION TITLE
Server Sent Events transport

### DISCUSSION THREAD

**Discussion by tonyrewin** [mirumee/ariadne#991] [Ideas]
I want to implement the subject for graphql subscriptions. What is the best way to do that? I tried to rewrite **GraphQLTransportWS** class and then pass this object to **websocket_handler** parameter of ariadne's GraphQL(..) init. Am I going well? What would you recommend?

**Comment by rafalp** [comment]
Hey. Websocket handler is called for websocket protocol. So its different thing than Server Sent Event. I don't have knowledge of SSE to say more about this, but quick google shows that The Guild provides dedicated handler for SSE connections, so perhaps we should provide `sse_handler` on the `GraphQL` app to use in this case?

**Reply by tonyrewin** [reply]
That would be awesome and I've tried to write it by myself, but found WSHandler and HTTPHandler only is there. Not sure if they are useful as base for SSEHandler code

**Comment by rafalp** [comment]
HTTP handler is closer to what you are trying to accomplish because its HTTP bases. From my understanding SSE handler should use Ariadne's `subscribe` instead of `graphql` to run query, and return a HTTP2 response on every event that resolves to not null?

**Reply by tonyrewin** [reply]
Yes, sure, SSE is HTTP itself, but it is not so obvious how to use it for GraphQL subscription in ariadne. If it is not necessary to implement the separate sse_handler= parameter in API can you please provide an example at least.

**Reply by tonyrewin** [reply]
Is there any method to reuse HTTPHandler code for graphql subscriptions or recommendations how to deal with EventSource better way?

**Reply by cancan101** [reply]
FYI, SSE does not require HTTP/2.

I think the rough approach here would be to pull code like this:
https://github.com/mirumee/ariadne/blob/9b55730e2c357a12e4079dee5359ae55d8052312/ariadne/asgi/handlers/graphql_ws.py#L201-L217
into here on the http handler:
https://github.com/mirumee/ariadne/blob/9b55730e2c357a12e4079dee5359ae55d8052312/ariadne/asgi/handlers/http.py#L146-L167

**Comment by cancan101** [comment]
Here is the RFC for this protocol: https://github.com/graphql/graphql-over-http/blob/main/rfcs/GraphQLOverSSE.md
